drivers/at86rf215/at86rf215.c

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_at86rf215
+ * @{
+ *
+ * @file
+ * @brief       Implementation of public functions for AT86RF215 driver
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ * @}
+ */
+
+
+#include "luid.h"
+#include "byteorder.h"
+#include "net/ieee802154.h"
+#include "net/gnrc.h"
+#include "unaligned.h"
+#include "at86rf215_internal.h"
+#include "at86rf215_netdev.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static void _setup_interface(at86rf215_t *dev, const at86rf215_params_t *params)
+{
+    netdev_t *netdev = (netdev_t *)dev;
+
+    netdev->driver = &at86rf215_driver;
+    dev->params = *params;
+    dev->state = AT86RF215_STATE_OFF;
+}
+
+void at86rf215_setup(at86rf215_t *dev_09, at86rf215_t *dev_24, const at86rf215_params_t *params)
+{
+    /* configure the sub-GHz interface */
+    if (dev_09) {
+        dev_09->RF = &RF09_regs;
+        dev_09->BBC = &BBC0_regs;
+        _setup_interface(dev_09, params);
+        dev_09->sibling = dev_24;
+    }
+
+    /* configure the 2.4 GHz interface */
+    if (dev_24) {
+        dev_24->RF = &RF24_regs;
+        dev_24->BBC = &BBC1_regs;
+        _setup_interface(dev_24, params);
+        dev_24->sibling = dev_09;
+    }
+}
+
+void at86rf215_reset_cfg(at86rf215_t *dev)
+{
+    netdev_ieee802154_reset(&dev->netdev);
+
+    /* set device address */
+    luid_get_short((network_uint16_t *)&dev->netdev.short_addr);
+    luid_get_eui64((eui64_t *)&dev->netdev.long_addr);
+
+    if (is_subGHz(dev)) {
+        dev->netdev.chan = AT86RF215_DEFAULT_SUBGHZ_CHANNEL;
+    } else {
+        dev->netdev.chan = AT86RF215_DEFAULT_CHANNEL;
+    }
+
+    dev->netdev.pan = IEEE802154_DEFAULT_PANID;
+
+    /* set default options */
+    dev->retries_max = 3;
+    dev->csma_retries_max = 4;
+    dev->csma_maxbe = 5;
+    dev->csma_minbe = 3;
```

Comment:
Maybe use defines for these defaults?

Reply:
Are there some defaults for that in RIOT already?
I'll check out what the other drivers are doing.

Reply:
`at86rf2xx` also just does `at86rf2xx_set_csma_max_retries(dev, 4)` - I've added some defines.